model/host/db.go

```diff
@@ -616,8 +616,36 @@ func FindStaleRunningTasks(cutoff time.Duration) ([]task.Task, error) {
 	return tasks, nil
 }
 
-// LastCommunicationTimeElapsed returns hosts which have never communicated or have not communicated in too long.
-func LastCommunicationTimeElapsed(currentTime time.Time) bson.M {
+// AgentLastCommunicationTimeElapsed finds legacy hosts which do not have an
+// agent or whose agents have not communicated recently.
+func AgentLastCommunicationTimeElapsed(currentTime time.Time) bson.M {
+	bootstrapKey := bsonutil.GetDottedKeyName(DistroKey, distro.BootstrapMethodKey)
+	cutoffTime := currentTime.Add(-MaxLCTInterval)
+	return bson.M{
+		StatusKey:        evergreen.HostRunning,
+		StartedByKey:     evergreen.User,
+		HasContainersKey: bson.M{"$ne": true},
```

Comment:
can this be "false" just for clarity?

Reply:
I took most of these keys from `LastCommunicationTimeElapsed()`, which also does this. Since `Host.HasContainers` has `omitempty` on the struct tags, a host might not have the HasContainers key. Using `{$ne: true}` should check for both false and a nonexistent key.